src/routers/management/config.py

```diff
@@ -23,12 +23,13 @@ def update_config(
     # Update config
     new_config.to_runtime_env()
     new_config.to_env_file()
-    # If the new config has a different model provider,
+    # If the new config has a different model provider
+    # Or the AI config has not been configured yet
     # We need to:
     # 1. Reload the llama_index settings
-    # 2. Generate the index again
+    # 2. Reset the index
     init_settings()
-    if new_config.model_provider != config.model_provider:
+    if (new_config.model_provider != config.model_provider) or not config.configured:
```

Comment:
this is a general bug, i guess?

Reply:
I add this to automatically create the index (by calling reset index) for the first time you start the app. Let me check again; I'll remove it if it's not needed anymore.